### PR TITLE
[QC-357] Follow the change of the Data Sampling directory and namespace

### PR DESCRIPTION
**Pull request by knopers8**
This prepares QC for the change of the Data Sampling directory and namespace. It supports both versions - the current one and after the change, so it should be painless if done in right order.

### CODE REVIEW

File: Framework/CMakeLists.txt [hunk 1]
```diff
@@ -95,6 +95,7 @@ target_link_libraries(QualityControl
                              ms_gsl::ms_gsl
                              QualityControlTypes
                              O2::Mergers
+                             O2::DataSampling
```
Comment (knopers8) [inline Framework/CMakeLists.txt:98]: @Barthelemy Is it possible to link the library, but do not complain if it is not found?

Comment (Barthelemy) [reply Framework/CMakeLists.txt:98]: yes, with generator expressions: https://cmake.org/cmake/help/v3.2/manual/cmake-generator-expressions.7.html

Comment (Barthelemy) [reply Framework/CMakeLists.txt:98]: Probably something like `"$<$<O2::DataSampling>:O2::DataSampling>"` (I don't remember exactly)

**Comment by Barthelemy** [comment]
Should I review or wait ?

File: Framework/CMakeLists.txt (hunk 1 above)
Comment (knopers8) [reply Framework/CMakeLists.txt:98]: Thanks, probably `$<$<TARGET_EXISTS:O2::DataSampling>:O2::DataSampling>` should work. I am giving it a try now

**Comment by knopers8** [comment]
I will fix linking to DataSampling first, no need for the review yet.

**Comment by knopers8** [comment]
It should be OK now, you can review.

**Review by Barthelemy — APPROVED** [review]
All good. We should point users to this PR when they are going to make breaking chagnes to O2.

**Pull request merged**